tests/test_resolver.py

```diff
@@ -65,3 +65,282 @@ def test_provider_choose_sdist():
+  "trees_url": "https://api.github.com/repos/python-wheel-build/fromager/git/trees{/sha}",
+  "statuses_url": "https://api.github.com/repos/python-wheel-build/fromager/statuses/{sha}",
+  "languages_url": "https://api.github.com/repos/python-wheel-build/fromager/languages",
+  "stargazers_url": "https://api.github.com/repos/python-wheel-build/fromager/stargazers",
+  "contributors_url": "https://api.github.com/repos/python-wheel-build/fromager/contributors",
+  "subscribers_url": "https://api.github.com/repos/python-wheel-build/fromager/subscribers",
+  "subscription_url": "https://api.github.com/repos/python-wheel-build/fromager/subscription",
+  "commits_url": "https://api.github.com/repos/python-wheel-build/fromager/commits{/sha}",
+  "git_commits_url": "https://api.github.com/repos/python-wheel-build/fromager/git/commits{/sha}",
+  "comments_url": "https://api.github.com/repos/python-wheel-build/fromager/comments{/number}",
+  "issue_comment_url": "https://api.github.com/repos/python-wheel-build/fromager/issues/comments{/number}",
+  "contents_url": "https://api.github.com/repos/python-wheel-build/fromager/contents/{+path}",
+  "compare_url": "https://api.github.com/repos/python-wheel-build/fromager/compare/{base}...{head}",
+  "merges_url": "https://api.github.com/repos/python-wheel-build/fromager/merges",
+  "archive_url": "https://api.github.com/repos/python-wheel-build/fromager/{archive_format}{/ref}",
+  "downloads_url": "https://api.github.com/repos/python-wheel-build/fromager/downloads",
+  "issues_url": "https://api.github.com/repos/python-wheel-build/fromager/issues{/number}",
+  "pulls_url": "https://api.github.com/repos/python-wheel-build/fromager/pulls{/number}",
+  "milestones_url": "https://api.github.com/repos/python-wheel-build/fromager/milestones{/number}",
+  "notifications_url": "https://api.github.com/repos/python-wheel-build/fromager/notifications{?since,all,participating}",
+  "labels_url": "https://api.github.com/repos/python-wheel-build/fromager/labels{/name}",
+  "releases_url": "https://api.github.com/repos/python-wheel-build/fromager/releases{/id}",
+  "deployments_url": "https://api.github.com/repos/python-wheel-build/fromager/deployments",
+  "created_at": "2024-05-31T15:49:02Z",
+  "updated_at": "2024-06-26T14:52:13Z",
+  "pushed_at": "2024-06-26T14:52:09Z",
+  "git_url": "git://github.com/python-wheel-build/fromager.git",
+  "ssh_url": "git@github.com:python-wheel-build/fromager.git",
+  "clone_url": "https://github.com/python-wheel-build/fromager.git",
+  "svn_url": "https://github.com/python-wheel-build/fromager",
+  "homepage": "https://pypi.org/project/fromager/",
+  "size": 907,
+  "stargazers_count": 1,
+  "watchers_count": 1,
+  "language": "Python",
+  "has_issues": true,
+  "has_projects": true,
+  "has_downloads": true,
+  "has_wiki": false,
+  "has_pages": false,
+  "has_discussions": false,
+  "forks_count": 4,
```

Comment:
This will fail with more forks yes?

Reply:
This is a mock response. It's used to provide a fake API endpoint so that we always get consistent data, and so the tests don't have to talk to GitHub.com. Most of the response isn't actually used, but it's needed to make the GitHub client work properly so I included it all.